Try wrapping entire coming soon page under cover block to fix layout issue

ilyasfoo opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Another attempt to partially fix layout issues from https://github.com/woocommerce/team-ghidorah/issues/311 by wrapping everything under the cover block and flexbox positioning instead of using absolute positioning.

_Note that this does not yet fix the styling issues (such as changing background color, default margin/padding mismatches, etc.)_

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

1. Enable `launch-your-store` feature flag.
2. Go to `WooCommerce > Settings > Site Visibility` and enable `Coming soon` checkbox and save.
3. Go to `Appearance > Editor > Templates > Page: Coming soon` and edit it.
4. Try adding blocks around and see if the frontend layout matches.

https://github.com/woocommerce/woocommerce/assets/3747241/f9522cca-24d0-4829-b0d0-217b0a7e7302


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

ilyasfoo committed:
Refactor to wrap everything under cover block

ilyasfoo committed:
Changelog

ilyasfoo committed:
Cleanup and fix classic theme

ilyasfoo committed:
Wrap center with stack for ease of adding elements nearby

ilyasfoo commented on plugins/woocommerce-blocks/assets/js/blocks/coming-soon/edit.tsx:
```diff
@@ -118,16 +115,23 @@ export default function Edit( { attributes, setAttributes } ) {
+                    flex-direction: column;
+                    justify-content: space-between;
+                }
+                .woocommerce-coming-soon-banner-container > .wp-block-group__inner-container {
```
This is required for classic themes, `wp-block-group__inner-container` appears out of nowhere 🤷

ilyasfoo commented on plugins/woocommerce/patterns/coming-soon-entire-site.php:
```diff
@@ -156,15 +173,12 @@
-.woocommerce-coming-soon-header, .coming-soon-cover .wp-block-cover__background {
-	background-color: #bea0f2 !important;
+.woocommerce-coming-soon-header {
+	height: 40px;
```
Small hack: Fixed header height in order to ensure small modifications will keep the center part relatively stable

psealock commented on plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx:
```diff
@@ -135,15 +139,12 @@ registerBlockType( metadata, {
+                    align-items: stretch;
                 }
-                .woocommerce-coming-soon-header, .coming-soon-cover .wp-block-cover__background {
-                    background-color: ${ color } !important;
```
With the Coming soon block selected, the background color is not longer respecting the selection

<img width="289" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/8addb2a8-07cf-4178-805c-1c31fb19728a">

ilyasfoo replied:
Good catch!

ilyasfoo replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/46914/commits/c471e520d322b7073fa4ae854d7113be0b7c92a5

psealock reviewed (commented):
This is working nicely @ilyasfoo.

I have [a bit of a refactor currently under review](https://github.com/woocommerce/woocommerce/pull/46854) as well which will cause some conflicts. Do you want to give it a shot here or after merge?

ilyasfoo committed:
: 1708591425:0;wc-cli admin build:watch
Merge remote-tracking branch 'origin/trunk' into fix/lys-attempt-using-cover-block-fix

ilyasfoo commented:
> I have https://github.com/woocommerce/woocommerce/pull/46854 as well which will cause some conflicts. Do you want to give it a shot here or after merge?

@psealock  Thanks for the heads up! I think it'll be better to handle after the PR is merged. Would you have that merged soon? I think it looks good to merge with 1 review

psealock commented:
I merged https://github.com/woocommerce/woocommerce/pull/46854 @ilyasfoo, let me know if any issues arise

ilyasfoo committed:
Merge remote-tracking branch 'origin/trunk' into fix/lys-attempt-using-cover-block-fix

ilyasfoo committed:
Update styles

ilyasfoo committed:
Fix dynamic background color

ilyasfoo committed:
Style clean up

ilyasfoo commented:
> I merged https://github.com/woocommerce/woocommerce/pull/46854 @ilyasfoo, let me know if any issues arise

@psealock Thanks! I've resolved conflicts without issues

psealock reviewed (approved):
This is looking great @ilyasfoo, testing well and code changes look good 🚢

chihsuan reviewed (approved):
Thanks @ilyasfoo. Nice fix.  🚢
